pkg/agent/flags/flag_controller.go

```diff
@@ -101,29 +104,132 @@ func (fc *FlagController) notifyObservers(flagKeys ...keys.FlagKey) {
 	}
 }
 
+func (fc *FlagController) AutoloadedExtensions() []string {
+	return fc.cmdLineOpts.AutoloadedExtensions
+}
+
+func (fc *FlagController) SetKolideServerURL(url string) error {
+	return fc.setControlServerValue(keys.KolideServerURL, []byte(url))
+}
+func (fc *FlagController) KolideServerURL() string {
+	return NewStringFlagValue(
+		WithDefaultString(fc.cmdLineOpts.KolideServerURL),
+	).get(fc.getControlServerValue(keys.KolideServerURL))
+}
+
+func (fc *FlagController) SetKolideHosted(hosted bool) error {
+	return fc.setControlServerValue(keys.KolideHosted, boolToBytes(hosted))
+}
+func (fc *FlagController) KolideHosted() bool {
+	return NewBoolFlagValue(WithDefaultBool(false)).get(fc.getControlServerValue(keys.KolideHosted))
+}
+
+func (fc *FlagController) EnrollSecret() string {
+	return NewStringFlagValue(
+		WithDefaultString(fc.cmdLineOpts.EnrollSecret),
+	).get(nil)
+}
+
+func (fc *FlagController) EnrollSecretPath() string {
+	return NewStringFlagValue(
+		WithDefaultString(fc.cmdLineOpts.EnrollSecretPath),
+	).get(nil)
+}
+
+func (fc *FlagController) RootDirectory() string {
+	return NewStringFlagValue(
+		WithDefaultString(fc.cmdLineOpts.RootDirectory),
+	).get(nil)
+}
+
+func (fc *FlagController) OsquerydPath() string {
+	return NewStringFlagValue(
+		WithDefaultString(fc.cmdLineOpts.OsquerydPath),
+	).get(nil)
+}
+
+func (fc *FlagController) CertPins() [][]byte {
+	return fc.cmdLineOpts.CertPins
+}
+
+func (fc *FlagController) RootPEM() string {
+	return NewStringFlagValue(
+		WithDefaultString(fc.cmdLineOpts.RootPEM),
+	).get(nil)
+}
+
+func (fc *FlagController) SetLoggingInterval(interval time.Duration) error {
+	return fc.setControlServerValue(keys.LoggingInterval, durationToBytes(interval))
+}
+func (fc *FlagController) LoggingInterval() time.Duration {
+	return NewDurationFlagValue(fc.logger, keys.LoggingInterval,
+		WithDefault(fc.cmdLineOpts.LoggingInterval),
+		WithMin(5*time.Second),
+		WithMax(10*time.Minute),
+	).get(fc.getControlServerValue(keys.LoggingInterval))
+}
+
+func (fc *FlagController) EnableInitialRunner() bool {
+	return NewBoolFlagValue(
+		WithDefaultBool(fc.cmdLineOpts.EnableInitialRunner)).
+		get(nil)
+}
+
+func (fc *FlagController) Transport() string {
+	return NewStringFlagValue(
+		WithDefaultString(fc.cmdLineOpts.Transport),
+	).get(nil)
+}
+
+func (fc *FlagController) LogMaxBytesPerBatch() int {
+	return fc.cmdLineOpts.LogMaxBytesPerBatch
```

Comment:
Would we want to restrict this with a min/max?

Reply:
@directionless Do you have any strong feelings on the limits here? I don't have a good enough sense of what's reasonable here.

Reply:
Anything I know I recorded in https://github.com/kolide/launcher/blob/fb81fd8c680e6db53f6ebf045ef947a574e3fc80/cmd/launcher/extension.go#L69-L90

Does that help?